### DISCUSSION TITLE
Feature Request: Collaborative Playlists

### DISCUSSION THREAD

**Discussion by ecker00** [navidrome/navidrome#2770] [Ideas]
Allow adding collaborators to playlist which can see this playlist in their own library and add/remove songs. 

**Use cases:**
- Family/friends preparing a common playlist for party night
- Handy for parents with common playlist for children

Ref closed (stale) issue: #1523

**Comment by ecker00** [comment]
Just realized this is already possible, but without granular access control. Just make your playlist Public and it's visible to all users and everyone can add/remove content from it. For most, this approach will be plenty.

**Reply by DerRockWolf** [reply]
I have to double check this again but yesterday with the newest navidrome version this was not possible.
Did you maybe use an admin user to test this?

Jup, public playlists are only for visibility and don't allow all users to edit the playlist (including adding tracks).

**Comment by DerRockWolf** [comment]
@deluan would that be something you would be interested in having as an feature? If so I'm willing to contribute it :slightly_smiling_face:

**Reply by DerRockWolf** [reply]
In the meantime I've started to work on it.
Still quite a few things I need to do / find out.
Maybe I have a first working PR ready this weekend. (all commits nicely structured, so we could even merge in smaller parts for easier review 🙂)

A few details about the implementation:
- Users can be added to the playlist as either editor or viewer by the playlist owner.
- Editor can add / remove tracks to / from the playlist
- There is a new mapping table, similar to the `user_library` table. It holds the playlist to user relation including the permission.

**Reply by DerRockWolf** [reply]
A draft PR is already open since may: https://github.com/navidrome/navidrome/pull/5460

I'm willing to continue working on it and pushing it through.
